src/netpol_synth.py

```diff
@@ -154,33 +168,118 @@ def _xgress_conns_to_rules(conns, is_ingress):
 
         return res_rules
 
-    def synthesize(self, output_file):
-        """
-        Generates NetworkPolicies in yaml format based on the analysis done in the ctor.
-        If output file is specified, the output is dumped into the file. Otherwise, stdout is used
-        :param output_file: A file opened for writing
-        :return: None
-        """
+    def _find_deployments_from_pod_selector(self, selectors):
+        # selectors is of type list[LabelSelector]
+        assert all(isinstance(selector, LabelSelector) for selector in selectors)
+        res = []
+        for deploy in self.deployments.values():
+            labels = deploy.labels
+            if all(selector.matches(labels) for selector in selectors):
+                res.append(deploy)
+        return res
+
+    def _get_auth_policy_source_from_baseline_rule_selector(self, selectors):
+        # selectors is of type list[LabelSelector] or IpSelector
+        ip_blocks_list = []
+        res = {}
+        if isinstance(selectors, IpSelector):
+            ip_blocks_list = [str(selectors.ipn)]
+        else:
+            assert all(isinstance(selector, LabelSelector) for selector in selectors)
+            src_deployments = self._find_deployments_from_pod_selector(selectors)
+            res = self._gst_auth_policy_source_from_deployments(src_deployments)
+        if ip_blocks_list:
+            res['ipBlocks'] = ip_blocks_list
+        return res
```

Comment:
```suggestion
        if isinstance(selectors, IpSelector):
            return {'ipBlocks': [str(selectors.ipn)]}

        assert all(isinstance(selector, LabelSelector) for selector in selectors)
        src_deployments = self._find_deployments_from_pod_selector(selectors)
        return self._gst_auth_policy_source_from_deployments(src_deployments)
```
The `from` or `to` selectors of a given baseline rule can not mix ipBlock and LabelSelector

Reply:
Also, better have a test for using a Baseline Rule with IPBlcok